colour/blindness/machado2010.py

```diff
@@ -0,0 +1,271 @@
+    WS, YB, RG = tsplit(WSYBRG)
+
+    primaries = primaries.clone().align(cmfs.shape, left=0, right=0)
+    R, G, B = tsplit(primaries.values)
+
+    WS_R = np.trapz(R * WS, wavelengths)
+    WS_G = np.trapz(G * WS, wavelengths)
+    WS_B = np.trapz(B * WS, wavelengths)
+
+    YB_R = np.trapz(R * YB, wavelengths)
+    YB_G = np.trapz(G * YB, wavelengths)
+    YB_B = np.trapz(B * YB, wavelengths)
+
+    RG_R = np.trapz(R * RG, wavelengths)
+    RG_G = np.trapz(G * RG, wavelengths)
+    RG_B = np.trapz(B * RG, wavelengths)
+
+    M_G = np.array([[WS_R, WS_G, WS_B],
+                    [YB_R, YB_G, YB_B],
+                    [RG_R, RG_G, RG_B]])
+
+    PWS = 1 / (WS_R + WS_G + WS_B)
+    PYB = 1 / (YB_R + YB_G + YB_B)
+    PRG = 1 / (RG_R + RG_G + RG_B)
+
+    M_G *= np.array([PWS, PYB, PRG])[:, np.newaxis]
+
+    return M_G
+
+
+def anomalous_trichromacy_cmfs_Machado2010(cmfs, d_LMS):
+    """
+    Shifts given *LMS* cone fundamentals colour matching functions with given
+    :math:`\Delta_{LMS}` shift amount in nanometers to simulate anomalous
+    trichromacy.
+
+    Parameters
+    ----------
+    cmfs : LMS_ConeFundamentals
+        *LMS* cone fundamentals colour matching functions.
+    d_LMS : array_like
+        :math:`\Delta_{LMS}` shift amount in nanometers as an array of ints.
+
+    Notes
+    -----
+    -   Input *LMS* cone fundamentals colour matching functions steps size is
+        expected to be 1 nanometer, non complying input will be interpolated
+        at 1 nanometer steps size.
+    -   Input :math:`\Delta_{LMS}` shift amount is in domain [0, 20].
+
+    Returns
+    -------
+    LMS_ConeFundamentals
+        Anomalous trichromacy *LMS* cone fundamentals colour matching
+        functions.
+
+    Examples
+    --------
+    >>> from colour import LMS_CMFS
+    >>> cmfs = LMS_CMFS.get('Stockman & Sharpe 2 Degree Cone Fundamentals')
+    >>> cmfs[450]
+    array([ 0.0498639,  0.0870524,  0.955393 ])
+    >>> anomalous_trichromacy_cmfs_Machado2010(cmfs, np.array([15, 0, 0]))[450]  # noqa  # doctest: +ELLIPSIS
+    array([ 0.0891288...,  0.0870524 ,  0.955393  ])
+    """
+
+    cmfs = cmfs.clone()
+    if cmfs.shape.steps != 1:
+        cmfs = cmfs.clone().interpolate(SpectralShape(steps=1))
+
+    L, M, S = tsplit(cmfs.values)
+    d_L, d_M, d_S = np.asarray(d_LMS).astype(np.int_)
+
+    wavelengths = cmfs.wavelengths
+    area_L = np.trapz(L, wavelengths)
+    area_M = np.trapz(M, wavelengths)
+
+    alpha = lambda x: (20 - x) / 20
+
+    # Corrected equations as per:
+    # http://www.inf.ufrgs.br/~oliveira/pubs_files/
+    # CVD_Simulation/CVD_Simulation.html#Errata
+    L_a = alpha(d_L) * L + 0.96 * area_L / area_M * (1 - alpha(d_L)) * M
+    M_a = alpha(d_M) * M + 1 / 0.96 * area_M / area_L * (1 - alpha(d_M)) * L
+    # TODO: Check inconsistency with ground truth values, d_S domain seems
+    # to be [5, 59] instead of [0, 20].
+    S_a = cmfs.s_bar.clone().shift(d_S).values
```

Comment:
The main issue in our implementation lies probably here:
-  The shift to somehow match the dataset *Tritanomaly* matrices seems to be [5, 59] instead of [0, 20], there is probably something going on with the fact the *Smith & Pokorny 1975 Normal Trichromats* CMFS have 5 nanometers bins.
- Which leads me to the fact that I would like to find a way to offset any spd by an arbitrary offset, right now what I'm doing is interpolating the spectra to 1 nanometer bin and then shifting the indexes of the required nanometers amount but say that you want a 0.01nm shift, it would be quite expensive to interpolate the SPD at 0.01nm.